src/adv.py

```diff
@@ -49,3 +58,82 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+key_input = ""
+current_room = ""
+
+while key_input != "q":
+
+    for r in room:
+        if player1 in room[r].players:
+            print("{}: {}".format(
+                room[r].name, room[r].description))
+            room[r].print_contents()
+            player1.print_inventory()
+            current_room = r
+
+    key_input = input(
+        "Enter a direction [n, e, s, w] or a command [grab <item>]: ")
```

Comment:
Very clean implementation throughout the while loop! Solid input prompt as well.